onnxruntime/core/common/threadpool.cc

```diff
@@ -175,25 +197,43 @@ ThreadPool::ThreadPool(const std::string& name, int num_threads)
 
 void ThreadPool::Schedule(std::function<void()> fn) { impl_->Schedule(fn); }
 
-void ThreadPool::ParallelFor(int32_t total, std::function<void(int32_t)> fn) {
+void ThreadPool::ParallelFor(int64_t total, std::function<void(int64_t)> fn) {
   if (total <= 0) return;
 
-  if (total == 1) {
-    fn(0);
+  // If we have no need for threads, run sequentially
+  if (total == 1 || NumThreads() == 1) {
+    for (int i = 0; i < total; i++) {
+      fn(0);
+    }
```

Comment:
fn(i)?

Reply:
This looks like a bad merge.  Let me check it out.